src/main/java/com/flansmod/common/guns/ItemGun.java

```diff
@@ -282,6 +282,30 @@ public void addInformation(ItemStack stack, EntityPlayer player, List lines, boo
 
     @SideOnly(Side.CLIENT)
     private void onUpdateClient(ItemStack itemstack, World world, Entity entity, int i, boolean flag) {
+
+        if (Minecraft.getMinecraft().thePlayer != null && Minecraft.getMinecraft().thePlayer == entity
+                && Minecraft.getMinecraft().thePlayer.inventory.currentItem
+                != GunAnimations.lastInventorySlot) {
+            GunAnimations.lastInventorySlot = Minecraft.getMinecraft().thePlayer.inventory.currentItem;
+            ItemStack stack = Minecraft.getMinecraft().thePlayer.getHeldItem();
+            GunAnimations animations = FlansModClient.getGunAnimations((EntityLivingBase) entity, false);
+            if (stack != null && stack.getItem() instanceof ItemGun) {
+                float animationLength = ((ItemGun) stack.getItem()).type.switchDelay;
+                if (animationLength == 0) {
+                    animations.switchAnimationLength = animations.switchAnimationProgress = 0;
+                } else {
+                    animations.switchAnimationProgress = 1;
+                    animations.switchAnimationLength = animationLength;
+                    PlayerHandler
+                            .getPlayerData(Minecraft.getMinecraft().thePlayer, Side.CLIENT).shootTimeRight = Math
+                            .max(PlayerHandler
+                                            .getPlayerData(Minecraft.getMinecraft().thePlayer, Side.CLIENT).shootTimeRight,
+                                    animationLength);
+                }
+
+            }
+        }
+
         if (entity instanceof EntityPlayer && ((EntityPlayer) entity).inventory.getCurrentItem() == itemstack) {
             //Get useful objects
```

Comment:
You can move all this code down to here - where the checks on the player are happening, already